dags/utils/slack_operator.py

```diff
@@ -15,74 +18,125 @@ def get_central_time_exec_data(context):
 
 
 def task_fail_slack_alert_critical(context):
-    slack_webhook_token = BaseHook.get_connection(SLACK_CONN_ID).password
     slack_msg = """
             <!channel> :red_circle: Critical Failure
             *Task*: {task}  
-            *Dag*: {dag} 
+            *DAG*: {dag} 
             *Execution Time*: {exec_date}  
             *Log Url*: {log_url} 
             """.format(
         task=context.get("task_instance").task_id,
         dag=context.get("task_instance").dag_id,
-        ti=context.get("task_instance"),
         exec_date=get_central_time_exec_data(context),
         log_url=context.get("task_instance").log_url,
     )
     failed_alert = SlackWebhookOperator(
         task_id="slack_critical_failure",
-        http_conn_id="slack",
-        webhook_token=slack_webhook_token,
+        slack_webhook_conn_id=SLACK_CONN_ID,
         message=slack_msg,
         username="airflow",
     )
     return failed_alert.execute(context=context)
 
 
 def task_fail_slack_alert(context):
-    slack_webhook_token = BaseHook.get_connection(SLACK_CONN_ID).password
-    slack_msg = """
-            :red_circle: Task Failed. 
-            *Task*: {task}  
-            *Dag*: {dag} 
-            *Execution Time*: {exec_date}  
-            *Log Url*: {log_url} 
-            """.format(
-        task=context.get("task_instance").task_id,
-        dag=context.get("task_instance").dag_id,
-        ti=context.get("task_instance"),
-        exec_date=get_central_time_exec_data(context),
-        log_url=context.get("task_instance").log_url,
+
+    task_instance = context.get("task_instance")
+    task = context.get("task")
+    exception = context.get("exception")
+    exception_type = type(exception).__name__ if exception else "Unknown"
+    exception_message = (
+        str(exception) if exception else "No exception message available"
     )
+
+    # Extract additional information
+    dag = context.get("dag")
+    dag_id = task_instance.dag_id
+    task_id = task_instance.task_id
+    exec_date = get_central_time_exec_data(context)
+    log_url = task_instance.log_url
+    try_number = task_instance.try_number
+    max_tries = task.retries
+    operator = task.__class__.__name__  # Gets the operator class name
+    duration = getattr(task_instance, "duration", "Not available")
+
+    schedule_interval = dag.schedule_interval if dag else None
+
+    # Convert schedule_interval to human-readable format
+    if schedule_interval is None:
+        schedule_description = "None"
+    elif isinstance(schedule_interval, str):
+        try:
+            schedule_description = get_description(schedule_interval)
+        except Exception:
+            schedule_description = f"Cron expression: {schedule_interval}"
+    elif isinstance(schedule_interval, datetime.timedelta):
```

Comment:
Nice catch that schedules can come in [multiple data types](schedule_interval) 🙏 Might be good to extract this logic into a helper so it could have a docstring to go along with it.

Reply:
Good idea - I totally agree. https://github.com/cityofaustin/atd-airflow/commit/cf088a7faa108b349f5fd3e357c613d8e600d091 & https://github.com/cityofaustin/atd-airflow/commit/0cf80bb563a855dc39a448512ee2fe56b3cce3ca